### ISSUE TITLE
Execute JavaScript doesn't work before window load.

### ISSUE THREAD

**Issue by blainesch** [electron/electron#5331]
* Electron version: v0.37.7
* Operating system: OS X El Capitan (10.11.4)

Related to #5291 and #5319

Using the [electron quick start](https://github.com/electron/electron-quick-start) I'm trying to use the `executeJavaScript` method.

This works:

~~~ diff
diff --git a/main.js b/main.js
index a9b3175..5ce036b 100644
--- a/main.js
+++ b/main.js
@@ -17,6 +17,12 @@ function createWindow () {
   // and load the index.html of the app.
   mainWindow.loadURL('file://' + __dirname + '/index.html');
 
+  mainWindow.webContents.on('did-finish-load', function() {
+    mainWindow.webContents.executeJavaScript('10+2', function(answer) {
+      console.log('answer', answer); // 12
+    });
+  });
+
   // Open the DevTools.
   mainWindow.webContents.openDevTools();
~~~

but this does not:

~~~ diff
diff --git a/main.js b/main.js
index a9b3175..fe07bea 100644
--- a/main.js
+++ b/main.js
@@ -17,6 +17,10 @@ function createWindow () {
   // and load the index.html of the app.
   mainWindow.loadURL('file://' + __dirname + '/index.html');
 
+  mainWindow.webContents.executeJavaScript('10+2', function(answer) {
+    console.log('answer', answer); // never gets executed
+  });
+
   // Open the DevTools.
   mainWindow.webContents.openDevTools();
~~~

**Comment by paulcbetts** [comment]
Well yeah. Where do you think that JavaScript is going to get executed?

**Comment by zcbenz** [comment]
This has been fixed with #5319.
